app/src/main/java/com/example/keybindhelper/ui/Projects/ProjectsFragment.kt

```diff
@@ -26,85 +26,75 @@ import com.google.android.material.textfield.TextInputEditText
 
-
         val mainActivity=activity as MainActivity;
         mainActivity.showMenuItems(mainActivity.projectsFragmentActionMenuIds)
 
-        mainActivity.Menu?.findItem(R.id.action_add)?.setOnMenuItemClickListener {
+        mainActivity.menu?.findItem(R.id.action_add)?.setOnMenuItemClickListener {
             val pd=PromptDialog(root.context,"New Project Name","","",null);
             val projects= DatabaseManager.db.getProjects();
+
             pd.validation= PromptDialog.Validator {
                 ValidatorResponse(DatabaseManager.isProjectNameAvailable(projects,it),"A Project Already Exists By That Name")
             }
+
             pd.confirmedEvent= PromptDialog.ConfirmedEvent {
                 val p= Project();
                 p.name.value=it;
                 CurrentProjectManager.loadProject(p,false)
                 p.id=DatabaseManager.db.insert(p)
                 openKeybindFragment();
             }
+
             pd.ShowDialog()
             true;
         };
 
-
-
-
         val rv= root.findViewById<RecyclerView>(R.id.projects_recycler);
-        RefreshProjectList()
+        refreshProjectList()
         rv.layoutManager=LinearLayoutManager(rv.context)
 
         val searchEdit=root.findViewById<TextInputEditText>(R.id.projects_search_edit);
         searchEdit.addTextChangedListener(object :
             TextWatcher {
-            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
-
-            }
-
-            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-                RefreshProjectList();
+                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                    refreshProjectList();
+                }
+                override fun afterTextChanged(s: Editable?) {}
             }
-
-            override fun afterTextChanged(s: Editable?) {
-
-            }
-        }
         )
-
         return root;
-
     }
+
     fun openKeybindFragment(){
         NavHostFragment.findNavController(this).navigate(R.id.nav_keybind)
     }
-    fun RefreshProjectList():List<Project>{
+
+    fun refreshProjectList():List<Project>{
         val rv= root.findViewById<RecyclerView>(R.id.projects_recycler);
         val projects=DatabaseManager.getOrderedProjects()
         val searchBox=root.findViewById<EditText>(R.id.projects_search_edit);
-        if(searchBox.text.length>0) {
+        if(searchBox.text.isNotEmpty()) {
             val filteredP= mutableListOf<Project>();
             val search=searchBox.text.toString().lowercase();
-            for (project in projects) {
-                if(project.name.value!!.lowercase().contains(search))
-                filteredP+=(project);
+
+            projects.forEach {
+                if (it.name.value!!.lowercase().contains(search))
```

Comment:
Use let?.{} instead of force unwrap (!!) to handle nulls.
let is null safe.
Force unwrap can lead to a NullPointerException, which should be avoided.